unittests/data/type_traits_castxml.hpp

```diff
@@ -195,8 +195,8 @@ namespace no{
     typedef std::set< std::string > string_set_type;
     typedef std::multimap< std::string, std::string > s2s_multimap_type;
     typedef detail::vertex vertex_type;
-    typedef detail::const_item const_item_t;
-    typedef detail::const_container const_container_t;
+    // typedef detail::const_item const_item_t;
```

Comment:
I will need a little bit more time to think about it. I already tried to make some changes in is_const, for debugging purposes, and I also saw that changing it's behaviour had an effect on some tests.

I learned from some old tests that they were just added as a safeguard. The authors new that the output was dubious (often due to gccxml writing strange stuff in the xml files), but it was just a way to make sure nothing changed when the next gccxml update came (and if it changed, they new the xml output was different).

So these tests may be wrong, and could be removed. But I want to be sure of it first, that's why I will need to run the tests and have a look myself.

+1 on adding so many tests. There are already many; and they are really helpful when making changes.